examples/fizzbuzz/main.c

```diff
@@ -0,0 +1,227 @@
+#include <qmlbind.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+/*!
+ * \file main.c
+ * \brief This example shows how to wrap fizzbuzz into a metaclass with `libqmlbind`.
+ */
+
+#define UNUSED(x) (void)(x)
+
+#define MIN_RESULT_LEN 9 // len("FizzBuzz") + 1
+
+
+typedef struct fizzbuzz
+{
+    int input;
+    char *result;
+    int result_len;
+    qmlbind_signal_emitter *signalEmitter;
+} fizzbuzz_t;
+
+
+void on_changed(fizzbuzz_t *fb, const qmlbind_value *input)
+{
+    qmlbind_string *inputStr = qmlbind_value_get_string(input);
+    int input_str_len = qmlbind_string_get_length(inputStr);
+    qmlbind_string_release(inputStr);
+    if (input_str_len == 0)
+        return; // Ignore empty textfield
+
+    input_str_len++; // take '\0' into account
+    if (input_str_len > fb->result_len) {
+        fb->result_len =
+            input_str_len > MIN_RESULT_LEN ? input_str_len : MIN_RESULT_LEN;
+        fb->result = realloc(fb->result, sizeof(char) * fb->result_len);
+        if (fb->result == NULL) {
+            printf("Error reallocating memory");
+            qmlbind_application_exit(0);
```

Comment:
Of course @seanchas116 is right, sorry, I did not see the rest of the code that's executed in the overview.